hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -76,6 +80,7 @@ public MapQueryEngineImpl(MapServiceContext mapServiceContext) {
         this.partitionService = nodeEngine.getPartitionService();
         this.logger = nodeEngine.getLogger(getClass());
         this.queryResultSizeLimiter = new QueryResultSizeLimiter(mapServiceContext, logger);
+        this.queryOptimizer = QueryOptimizerFactory.newOptimizer(nodeEngine.getGroupProperties());
```

Comment:
I didn't want to include Optimizer creation logic inside MapQueryEngineImpl. 
I can pass it as an argument, I'm fine with that.